### PR TITLE
Simplify flow func augmentation logic in `connectivity` module

### PR DESCRIPTION
**Pull request by mohamedrezk122**
Following the steps of #7036 and #7166, I have refactored the if/elif statements in ``networkx/algorithms/connectivity/connectivity.py`` with match/case statements.

There was a couple of unwanted matches since they can be replaced with ``or`` , I guess they were there because of reference equality with ``is``, but matching with name makes it easier to group similar conditions together.

### CODE REVIEW

File: networkx/algorithms/connectivity/connectivity.py [hunk 1]
```diff
@@ -199,15 +199,12 @@ def local_node_connectivity(
...
-    elif flow_func is dinitz:
-        kwargs["cutoff"] = cutoff
-    elif flow_func is boykov_kolmogorov:
-        kwargs["cutoff"] = cutoff
+    match flow_func.__name__:
+        case "shortest_augmenting_path":
+            kwargs["cutoff"] = cutoff
+            kwargs["two_phase"] = True
+        case "edmonds_karp" | "dinitz" | "boykov_kolmogorov":
+            kwargs["cutoff"] = cutoff
```
Comment (dschult) [inline networkx/algorithms/connectivity/connectivity.py:207]: This is better than the existing code, but even better would be:
```python
kwargs["cutoff"] = cutoff
if flow_func is shortest_augmenting_path:
    kwargs["two_phase"] = True
```

Comment (mohamedrezk122) [reply networkx/algorithms/connectivity/connectivity.py:207]: I thought of this solution, and it is indeed elegant. But I guessed that the flow functions are not  exhaustively listed, like maybe there is some flow function that should not be matched to edit its kwargs.

**Comment by dschult** [comment]
Yes -- good thought!
And a user may have written their own function. so we shouldn't just raise if we don't have one of the included flow functions.   The history here is helpful (and not apparent from the code -- but maybe from the "blame" or comments like this one). Originally the cutoff parameter was only present for some of the flow functions. This code was written and some elif didn't have the cutoff parameter. Then there was an effort made to ensure all flow_functions supported a cutoff kwarg. That effort was successful. And this logic was updated but not simplified. 

I guess an even simpler way to handle this is to include `cutoff` in the kwargs dict in the previous line where it is defined. Then it is just one if statement.

Users who write their own functions already have to have the other kwargs at least in their signature (they can ignore them).  So I guess that would put `cutoff` into that category. They'd have to at least put it in their function signature (or have a ** parameter). They don't have to do anything with it though.

**Comment by mohamedrezk122** [comment]
I got it, I am going to make the edits in a commit.

**Comment by mohamedrezk122** [comment]
There was a problem, that the ``cutoff`` kwarg is not defined for  the ``preflow_push`` flow function. So adding it directly to the kwargs dict is not going to work, so I wrote some extra check to see if it is not  ``preflow_push``. I guess the code reads like English with fewer if/elif or match statements.

**Review by rossbar — APPROVED** [review]
The simplification of all the conditional logic seems like an improvement to me. I wonder if it's possible to use `functools.partial` to solve this problem? Anyways, +1 for this cleanup as-is!

**Review by dschult — APPROVED** [review]

**Pull request merged**